src/core-asyncpipeline/AsyncPipeline/AsyncPipeline.T/AsyncPipeline.T.cs

```diff
@@ -0,0 +1,25 @@
+#nullable enable
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System
+{
+    public readonly partial struct AsyncPipeline<T> : IEquatable<AsyncPipeline<T>>
+    {
+        private static CancellationToken CanceledToken() => new(canceled: true);
+
+        private readonly ValueTask<T> valueTask;
+
+        private readonly bool isCanceled;
```

Comment:
It seems `isCanceled` field is redundant.

I'd like to suggest to remove it and create a property kind of
`private bool IsCanceled => cancellationToken.IsCancellationRequested`
for use everywhere instead of direct `cancellationToken.IsCancellationRequested`